// This file is automatically generated by sila2lib.proto_builder.proto_builder version 0.2.0
// :generation date: 2019-10-05 17:59:19
// 
// ---- PLEASE DO NOT MODIFY MANUALLY !! ---

syntax = "proto3";
import "SiLAFramework.proto";
package sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1;

// Feature: Syringe Configuration Controller
//   
//               Provides syringe pump specific functions for configuration (i.e. the configuration of the syringe itself).
//   
service SyringeConfigurationController {
    // Set Syringe Parameters
    //   
    //               Set syringe parameters.
    //               If you change the syringe in one device, you need to setup the new syringe parameters to get proper
    //   conversion of flow rate und volume units.
    //   
    rpc SetSyringeParameters(sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.SetSyringeParameters_Parameters) returns (sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.SetSyringeParameters_Responses) {}
    // Inner Diameter
    //   Inner diameter of the syringe tube in millimetres.
    rpc Subscribe_InnerDiameter(sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_InnerDiameter_Parameters) returns (stream sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_InnerDiameter_Responses) {}
    // Max Piston Stroke
    //   The maximum piston stroke defines the maximum position the piston can be moved to before it slips out of the syringe
    //   tube. The maximum piston stroke limits the maximum travel range of the syringe pump pusher.
    rpc Subscribe_MaxPistonStroke(sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_MaxPistonStroke_Parameters) returns (stream sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_MaxPistonStroke_Responses) {}
    
}

// ----------------- Data Type definitions -----------------


// ------ Command Parameter and Response definitions -------
message SetSyringeParameters_Parameters {
    sila2.org.silastandard.Real InnerDiameter = 1;
    sila2.org.silastandard.Real MaxPistonStroke = 2;
}

message SetSyringeParameters_Responses {
    
}
message Subscribe_InnerDiameter_Parameters {
    // Empty message
}

message Subscribe_InnerDiameter_Responses {
    sila2.org.silastandard.Real InnerDiameter = 1;
}

message Subscribe_MaxPistonStroke_Parameters {
    // Empty message
}

message Subscribe_MaxPistonStroke_Responses {
    sila2.org.silastandard.Real MaxPistonStroke = 1;
}

// ----------------- Metadata Definitions ------------------
